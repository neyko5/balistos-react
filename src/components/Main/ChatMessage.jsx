// @flow

import React from 'react';
import styled from 'styled-components';

const Message = styled.div`
  margin-bottom: 3px;
  color: #3e414c;
  font-size: 12px;
`;
export const Author = styled.span`
  margin-right: 3px;
  font-weight: 700;
  color: ${props => (props.current ? '#9FA600' : '#3e414c')}
`;

const ChatMessage = (props: {
  username: string,
  message: any,
}) => (
  <Message>
    <Author
      current={props.message.user.username === props.username}
      title={new Date(props.message.createdAt).toDateString()}
    >{props.message.user.username}:
    </Author>
    <span>{props.message.message}</span>
  </Message>
);

export default ChatMessage;
