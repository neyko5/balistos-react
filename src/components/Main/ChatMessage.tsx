import React from 'react';
import styled from 'styled-components';
import { ChatMessageType } from '../../types/index';

const Message = styled.div`
    margin-bottom: 3px;
    color: #3e414c;
    font-size: 12px;
`;
export const Author = styled.span`
    margin-right: 3px;
    font-weight: 700;
    color: ${(props: AuthorProps) => (props.current ? '#9FA600' : '#3e414c')};
`;

interface AuthorProps {
    current: boolean;
}

const ChatMessage = (props: { user: any; message: ChatMessageType }) => (
    <Message>
        <Author
            current={props.message.creator.id === props.user.id}
            title={new Date(props.message.created_at).toDateString()}
        >
            {props.message.creator.name}:
        </Author>
        <span>{props.message.message}</span>
    </Message>
);

export default ChatMessage;
