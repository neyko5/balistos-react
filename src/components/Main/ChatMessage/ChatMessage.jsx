import React from 'react';
import PropTypes from 'prop-types';

const ChatMessage = props => (
  <div className="message">
    <span
      title={new Date(props.message.createdAt).toLocaleString()}
      className={`author ${props.message.user.username === props.username ? 'green' : 'grey'}`}
    >{props.message.user.username}:</span><span>{props.message.message}</span>
  </div>
    );

ChatMessage.propTypes = {
  username: PropTypes.string,
  message: PropTypes.shape({
    user: PropTypes.shape({
      username: PropTypes.string.isRequired,
    }).isRequired,
    message: PropTypes.string.isRequired,
    createdAt: PropTypes.string.isRequired,
  }).isRequired,
};

ChatMessage.defaultProps = {
  username: undefined,
};

export default ChatMessage;
