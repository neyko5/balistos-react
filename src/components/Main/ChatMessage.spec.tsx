import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import React from 'react';
import renderer from 'react-test-renderer';

import ChatMessage, { Author } from './ChatMessage';

Enzyme.configure({ adapter: new Adapter() });

test('chats from current user should have author be displayed in green', () => {
    const chatMessage = mount(
        <ChatMessage
            user={{ id: '1', name: 'Tester 1' }}
            message={{
                id: '1',
                creator: { id: '1', name: 'Tester' },
                message: 'text',
                created_at: '2017-05-11T18:18:36.000Z',
            }}
        />
    );
    expect(chatMessage.find(Author)).toHaveStyleRule('color', '#9FA600');
});

test('chats from other users chats should have author be displayed in default color', () => {
    const chatMessage = mount(
        <ChatMessage
            user={{ id: '1', name: 'Tester' }}
            message={{
                id: '1',
                creator: { id: '2', name: 'Tester 2' },
                message: 'text',
                created_at: '2017-05-11T18:18:36.000Z',
            }}
        />
    );
    expect(chatMessage.find(Author)).toHaveStyleRule('color', '#3e414c');
});

test('chats should be displayed in default color if user is logged out', () => {
    const chatMessage = mount(
        <ChatMessage
            user={{ id: '1', name: 'Tester 1' }}
            message={{
                id: '1',
                creator: { id: '2', name: 'Tester 2' },
                message: 'text',
                created_at: '2017-05-11T18:18:36.000Z',
            }}
        />
    );
    expect(chatMessage.find(Author)).toHaveStyleRule('color', '#3e414c');
});

test('chat should have title parameter set', () => {
    const chatMessage = mount(
        <ChatMessage
            user={{ id: '1', name: 'Tester' }}
            message={{
                id: '1',
                creator: { id: '2', name: 'Tester 2' },
                message: 'text',
                created_at: '2017-05-11T18:18:36.000Z',
            }}
        />
    );
    expect(true).toBe(true);
    expect(chatMessage.find(Author).prop('title')).not.toBeFalsy();
});

test('should display authors username', () => {
    const chatMessage = shallow(
        <ChatMessage
            user={{ id: '1', name: 'Tester 1' }}
            message={{
                id: '1',
                creator: { id: '1', name: 'Tester 1' },
                message: 'text',
                created_at: '2017-05-11T18:18:36.000Z',
            }}
        />
    );
    expect(chatMessage.contains('Tester 1')).toBe(true);
});

test('should display message content', () => {
    const testMessage = 'test message';
    const chatMessage = shallow(
        <ChatMessage
            user={{ id: '1', name: 'Tester 1' }}
            message={{
                id: '1',
                creator: { id: '1', name: 'Tester 1' },
                message: testMessage,
                created_at: '2017-05-11T18:18:36.000Z',
            }}
        />
    );
    expect(chatMessage.contains(testMessage)).toBe(true);
});

test('should render correctly', () => {
    const chatMessage = renderer.create(
        <ChatMessage
            user={{ id: '1', name: 'Tester 1' }}
            message={{
                id: '1',
                creator: { id: '1', name: 'Tester 1' },
                message: 'test message',
                created_at: '2017-05-11T18:18:36.000Z',
            }}
        />
    );
    const tree = chatMessage.toJSON();
    expect(tree).toMatchSnapshot();
});
