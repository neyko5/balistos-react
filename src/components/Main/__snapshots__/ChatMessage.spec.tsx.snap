// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
.c0 {
  margin-bottom: 3px;
  color: #3e414c;
  font-size: 12px;
}

.c1 {
  margin-right: 3px;
  font-weight: 700;
  color: #9FA600;
}

<div
  className="c0"
>
  <span
    className="c1"
    title="Thu May 11 2017"
  >
    Tester 1
    :
  </span>
  <span>
    test message
  </span>
</div>
`;
